Libs/GObject-2.0/Native/Classes/ObjectWrapper.cs

```diff
@@ -39,11 +52,16 @@ public static class ObjectWrapper
             return (T) ctor.Invoke(new object[] { handle, ownedRef });
         }
 
-        private static Type GetTypeFromHandle(IntPtr handle)
+        private static Type GetTypeFromInstance(IntPtr handle)
         {
             TypeInstance.Struct instance = Marshal.PtrToStructure<TypeInstance.Struct>(handle);
-            TypeClass.Struct classStruct = Marshal.PtrToStructure<TypeClass.Struct>(instance.GClass);
-            return new Type(classStruct.GType);
+            TypeClass.Struct klass = Marshal.PtrToStructure<TypeClass.Struct>(instance.GClass);
+            var typeid = klass.GType;
+
+            if (typeid == 0)
```

Comment:
In which case do you think the type id could be zero? Would this even be an error for this method as a type id of 0 is valid for the `Type` struct.

If the check is necessary I would suggest checking for `Types.Invalid` as it explains the intend better.

Reply:
I think I got confused between `g_type_parent` returning zero and type itself being zero. In hindsight, this doesn't make sense - I'll remove it.